hicommonwealth/commonwealth
# User-Onboarding: Personal Information step -> Integrated api to store user consent for `Send me product updates and news` checkbox

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/6667

## Description of Changes
- Updated API to store user consent for product emails
- Integrated welcome modal "Send me product updates and news" checkbox with newer API changes 

## "How We Fixed It"
N/A

## Test Plan
- Enable `FLAG_USER_ONBOARDING_ENABLED`
- Run `yarn migrate-db`
- Create a new profile, and wait for the welcome onboard modal to appear
- Open network tab
- Now, In the first step of modal `Personal Information`, verify that when you enable `Send me product updates and news` and submit, it sends `promotionalEmailsEnabled: true` in `/updateProfile/v2` payload. (atm testing it this way helps since its not read anywhere else)

## Deployment Plan
N/A

## Other Considerations
N/A
